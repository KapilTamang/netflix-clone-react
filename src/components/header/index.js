import React, { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
	Background,
	Container,
	Group,
	Logo,
	ButtonLink,
	Feature,
	FeatureCallOut,
	Text,
	Link,
	Image,
	Profile,
	Dropdown,
	Search,
	SearchIcon,
	SearchInput,
	PlayButton,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
	return bg ? (
		<Background {...restProps} data-testid="header-bg">
			{children}
		</Background>
	) : (
		children
	);
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
	return <Feature {...restProps}>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
	children,
	...restProps
}) {
	return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
	return <PlayButton {...restProps}>{children}</PlayButton>;
};
Header.Text = function HeaderText({ children, ...restProps }) {
	return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
	return <Link {...restProps}>{children}</Link>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
	return <Container {...restProps}>{children}</Container>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
	return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
	return (
		<ReactRouterLink to={to}>
			<Logo {...restProps} />
		</ReactRouterLink>
	);
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
	return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Image = function HeaderImage({ src, children, ...restProps }) {
	return (
		<Image {...restProps} src={`/images/users/${src}.png`}>
			{children}
		</Image>
	);
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
	return <Profile {...restProps}>{children}</Profile>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
	return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Search = function HeaderSearch({
	searchTerm,
	setSearchTerm,
	...restProps
}) {
	const [searchActive, setSearchActive] = useState(false);

	return (
		<Search {...restProps}>
			<SearchIcon
				onClick={() => setSearchActive(!searchActive)}
				data-testid="search-click"
			>
				<img
					src="/images/icons/search.png"
					alt="Search"
					data-testid="search-icon"
				/>
			</SearchIcon>
			<SearchInput
				value={searchTerm}
				onChange={(e) => setSearchTerm(e.target.value)}
				placeholder="Search films and series"
				active={searchActive}
				data-testid="search-input"
			></SearchInput>
		</Search>
	);
};
