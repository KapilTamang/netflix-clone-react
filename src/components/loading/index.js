import React from 'react';
import { Spinner, LockBody, ReleaseBody, Image } from './styles/loading';

export default function Loading({ src, restProps }) {
	return (
		<Spinner {...restProps}>
			<LockBody />
			<Image src={`/images/users/${src}.png`} data-testid="loading-image" />
		</Spinner>
	);
}

Loading.ReleaseBody = function LoadingReleaseBody() {
	return <ReleaseBody />;
};
