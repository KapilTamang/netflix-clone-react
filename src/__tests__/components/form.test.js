import React from 'react';
import { render } from '@testing-library/react';
import { Form } from '../../components';

jest.mock('react-router-dom');

describe('<Form/>', () => {
	it('renders the <Form/> with populated data', () => {
		const { container, getByText, getByPlaceholderText } = render(
			<Form>
				<Form.Title>Sign In Now</Form.Title>
				<Form.Base>
					<Form.Input
						type="email"
						placeholder="Email or phone number"
						onChange={() => {}}
					/>
					<Form.Input
						type="password"
						placeholder="Password"
						onChange={() => {}}
					/>
					<Form.Submit type="submit" disabled>
						Sign In
					</Form.Submit>
					<Form.Text>
						New to Netflix?
						<Form.Link to="/signup">Sign up now.</Form.Link>
						<Form.TextSmall>
							This page is protected by Google reCAPTCHA to ensure you're not a
							bot. Learn more.
						</Form.TextSmall>
					</Form.Text>
				</Form.Base>
			</Form>
		);

		expect(
			getByText(
				"This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more."
			)
		).toBeTruthy();
		expect(getByText('Sign In Now')).toBeTruthy();
		expect(getByText('Sign In')).toBeTruthy();
		expect(getByPlaceholderText('Email or phone number')).toBeTruthy();
		expect(getByPlaceholderText('Password')).toBeTruthy();
		expect(getByText('Sign In').disabled).toBeTruthy();

		expect(container.firstChild).toMatchSnapshot();
	});

	it('renders <Form/> with an error', () => {
		const { container, getByText, queryByText } = render(
			<Form>
				<Form.Error>Your email address is already being used</Form.Error>
				<Form.Submit>Sign In</Form.Submit>
			</Form>
		);

		expect(getByText('Your email address is already being used')).toBeTruthy();
		expect(queryByText('Sing In')).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
