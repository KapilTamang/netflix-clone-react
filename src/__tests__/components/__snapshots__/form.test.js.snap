// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders <Form/> with an error 1`] = `
<div
  class="form__Container-sc-1uy70tk-0 fEJwlO"
>
  <div
    class="form__Error-sc-1uy70tk-1 gtcyCC"
  >
    Your email address is already being used
  </div>
  <button
    class="form__Submit-sc-1uy70tk-8 hJKpmN"
  >
    Sign In
  </button>
</div>
`;
